src/store/movies.ts

```diff
@@ -2,16 +2,27 @@ import { defineStore } from 'pinia';
 
 import { getAllMovies } from '@/api/movies';
 
+import type { Movie } from '@/types/movie';
+
+interface State {
+  allMovies: Movie[] | [];
+}
+
 export const useMoviesStore = defineStore({
   id: 'moviesStore',
   state: () => {
     return {
       allMovies: []
-    };
+    } as State;
   },
   getters: {
-    genres() {
-      return [...new Set(this.allMovies.map((movie) => movie.genre.name))];
+    genres(): string[] {
+      return [
+        ...new Set(this.allMovies.map((movie: Movie) => movie.genre.name))
+      ];
+    },
+    titles(): string[] {
+      return [...new Set(this.allMovies.map((movie: Movie) => movie.title))];
```

Comment:
I think TypeScript will infer these properly too 🤔 
If the State is typed out properly - everything else should work fine 🤔 
```suggestion
    genres() {
      return [
        ...new Set(this.allMovies.map((movie) => movie.genre.name))
      ];
    },
    titles() {
      return [...new Set(this.allMovies.map((movie) => movie.title))];
```

Reply:
But... I'm getting a TS error like this:
![image](https://user-images.githubusercontent.com/110823711/190627660-5b2c98fa-0c70-4abd-a3fe-5f4fd96ec3a7.png)

Does it mean that there is something wrong with the State typing?

Reply:
Hmmm the `allMovies` should be typed properly 🤔
Let me get back to you on this one as I'll try to see what's up with that

Reply:
Okay I think that you need the return type suggestion (for example `titles(): string { ... }`) - for some reason typescript seems to have a problem to infer the type here so it should be fine to leave it like that 👍 (the map() doesn't need a type suggestion though)

Reply:
Q answered on summary --> resolving :smiley: